src/keep_gpu/single_gpu_controller/rocm_gpu_controller.py

```
@@ -95,21 +99,42 @@ def _query_utilization(self) -> Optional[int]:
     def _keep_loop(self) -> None:
         torch.cuda.set_device(self.rank)
         tensor = None
+        attempts = 0
         while not self._stop_evt.is_set():
             try:
+                num_elements = int(self.vram_to_keep)
+                if num_elements <= 0:
+                    raise ValueError("vram_to_keep must be positive")
```

Ruff (0.15.0)

[warning] Abstract `raise` to an inner function

(TRY301)